app/src/org/commcare/connect/network/ApiConnectId.java

```diff
@@ -0,0 +1,400 @@
+package org.commcare.connect.network;
+
+import android.content.Context;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import org.commcare.CommCareApplication;
+import org.commcare.android.database.connect.models.ConnectLinkedAppRecord;
+import org.commcare.connect.ConnectConstants;
+import org.commcare.connect.ConnectDatabaseHelper;
+import org.commcare.android.database.connect.models.ConnectUserRecord;
+import org.commcare.core.network.AuthInfo;
+import org.commcare.dalvik.R;
+import org.commcare.preferences.HiddenPreferences;
+import org.commcare.preferences.ServerUrls;
+import org.commcare.utils.FirebaseMessagingUtil;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.services.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Date;
+import java.util.HashMap;
+
+public class ApiConnectId {
+    private static final String API_VERSION_NONE = null;
+    private static final String API_VERSION_CONNECT_ID = "1.0";
+
+    public static void linkHqWorker(Context context, String hqUsername, String hqPassword, String connectToken) {
+        String seatedAppId = CommCareApplication.instance().getCurrentApp().getUniqueId();
+        ConnectLinkedAppRecord appRecord = ConnectDatabaseHelper.getAppData(context, seatedAppId, hqUsername);
+        if (appRecord != null && !appRecord.getWorkerLinked()) {
```

Comment:
The only caller to this function (coming in a future PR) already retrieves the `appRecord` and checks whether the worker is linked before making this call, so I simplified the logic in this [commit](https://github.com/dimagi/commcare-android/pull/2847/commits/906137dddda08bff6efd751db158a01e8209306b) to take the `ConnectLinkedAppRecord` as an input instead of retrieving it again, and simply make the API call (rather than performing additional logic that the caller should handle).